ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -62,31 +68,12 @@ QObject {
         filters: ValueFilter { roleName: "isTest"; value: root.swapStore.areTestNetworksEnabled }
     }
 
-    ModelEntry {
-        id: fromTokenEntry
-        sourceModel: root.walletAssetsStore.walletTokensStore.plainTokensBySymbolModel
-        key: "key"
-        value: root.swapFormData.fromTokensKey
-    }
-
-    ModelEntry {
-        id: toTokenEntry
-        sourceModel: root.walletAssetsStore.walletTokensStore.plainTokensBySymbolModel
-        key: "key"
-        value: root.swapFormData.toTokenKey
-    }
-
-    ModelEntry {
-        id: selectedAccountEntry
-        sourceModel: root.nonWatchAccounts
-        key: "address"
-        value: root.swapFormData.selectedAccountAddress
-    }
-
     QtObject {
         id: d
 
         property string uuid
+        // storing txHash to verify against tx completed event
+        property string txHash
```

Comment:
strictly, `approvalTxHash`

Reply:
not necessarily, this will handle any tx done from swap? swap or approve.